lib/chef/util/path_helper.rb

```diff
@@ -142,6 +142,64 @@ def self.escape_glob(*parts)
       def self.relative_path_from(from, to)
         pathname = Pathname.new(Chef::Util::PathHelper.cleanpath(to)).relative_path_from(Pathname.new(Chef::Util::PathHelper.cleanpath(from)))
       end
+
+      # Retrieves the "home directory" of the current user while trying to ascertain the existence
+      # of said directory.  The path returned uses / for all separators (the ruby standard format).
+      # If a set of path elements may be provided.  They are appended as-is to the home path if the
+      # homepath exists.  If the home directory doesn't exist or an error is otherwise encountered,
+      # nil is returned.  If an optional block is provided, the joined path is passed to that block
+      # if the home path is valid and the result of the block is returned.
+      def self.home(*args)
+        @@home_dir ||= self.all_homes { |p| break p }
```

Comment:
Doh, it is a class method, I think @@ is fine.

The only danger I see from caching it is if the chef-client process is long running and the home directory is going to change.  That doesn't seem likely to me, but someone else may know better.